unic/ucd/tests/category_tests.rs

```diff
@@ -18,8 +18,112 @@ extern crate unic_utils;
 
 use unic_ucd::bidi::BidiClass;
 use unic_ucd::normal::is_combining_mark;
+use unic_ucd::category::GeneralCategory as GC;
 use unic_utils::iter_all_chars;
 
+/// `normal::is_combining_mark` and `GeneralCategory::is_mark()` are expected to return
+/// the same results.
+#[test]
+fn test_gen_cat_against_normal() {
+    for cp in iter_all_chars() {
+        // Every General_Category mark must be a combining mark
+        if GC::of(cp).is_mark() {
+            assert!(is_combining_mark(cp));
+        }
+
+        // Every combining mark must be a General_Category mark
+        if is_combining_mark(cp) {
+            assert!(GC::of(cp).is_mark());
+        }
+    }
+}
+
+/// `Bidi_Class=EN := General_Category in { N (Number) }`
+///
+/// <http://www.unicode.org/reports/tr9/#EN>
+#[test]
+fn test_bidi_en_against_gen_cat() {
+    // Every BC::EuropeanNumber must satisfy GC::is_number()
+    for cp in iter_all_chars() {
+        if BidiClass::of(cp) == BidiClass::EuropeanNumber {
+            assert!(GC::of(cp).is_number());
+        }
+    }
+}
+
+/// `Bidi_Class=ES := General_Category in { Sm (MathSymbol), Pd (DashPunctuation) }`
+///
+/// <http://www.unicode.org/reports/tr9/#ES>
+#[test]
+fn test_bidi_es_against_gen_cat() {
+    // Every BC::EuropeanSeparator must be a GC::MathSymbol, GC::DashPunctuation
+    for cp in iter_all_chars() {
+        if BidiClass::of(cp) == BidiClass::EuropeanSeparator {
+            assert!(
+                GC::of(cp) == GC::MathSymbol ||
+                GC::of(cp) == GC::DashPunctuation
+            );
+        }
+    }
+}
+
+/// `Bidi_Class=ET := General_Category in { S (Symbol), Cn (Unassigned),
+///                                         Po (OtherPunctuation) }`
+///
+/// <http://www.unicode.org/reports/tr9/#ET>
+#[test]
+fn test_bidi_et_against_gen_cat() {
+    // Every BC::EuropeanTerminator must satisfy GC::is_symbol() or
+    // be GC::Unassigned or GC::OtherPunctuation
+    for cp in iter_all_chars() {
+        if BidiClass::of(cp) == BidiClass::EuropeanTerminator {
+            assert!(
+                GC::of(cp).is_symbol()          ||
+                GC::of(cp) == GC::Unassigned    ||
+                GC::of(cp) == GC::OtherPunctuation
+            );
+        }
+    }
+}
+
+/// `Bidi_Class=AN := General_Category in { Cf (Format), No (OtherNumber),
+///                                         Po (OtherPunctuation), Nd (DecimalNumber) }`
+///
+/// <http://www.unicode.org/reports/tr9/#AN>
+#[test]
+fn test_bidi_an_against_gen_cat() {
+    // Every BC::ArabicNumber must be a GC::Format, GC::OtherNumber, GC::OtherPunctuation,
+    // or GC::DecimalNumber
+    for cp in iter_all_chars() {
+        if BidiClass::of(cp) == BidiClass::ArabicNumber {
+            assert!(
+                GC::of(cp) == GC::Format            ||
+                GC::of(cp) == GC::OtherNumber       ||
+                GC::of(cp) == GC::OtherPunctuation  ||
```

Comment:
Instead of `||` and `==`, you can also use `matches!()` which has a shorter syntax, and the `of()` function would get called only once.